api/src/websocket/utils/items.ts

```diff
@@ -157,13 +141,23 @@ export async function getItemsPayload(
 	const query = subscription.query ?? {};
 	const service = getService(subscription.collection, { schema, accountability });
 
-	if (!event?.action) {
-		return await service.readByQuery(query);
-	} else if (event.action === 'create') {
-		return await service.readMany([event.key], query);
-	} else if (event.action === 'delete') {
-		return event.keys;
-	} else {
-		return await service.readMany(event.keys, query);
+	if ('item' in subscription) {
+		if (event?.action === 'delete') {
+			// return only the subscribed id in case a bluk delete was done
+			return subscription.item;
+		} else {
+			return await service.readOne(subscription.item, query);
+		}
+	}
+
+	switch (event?.action) {
+		case undefined:
+			return await service.readByQuery(query);
+		case 'create':
+			return await service.readMany([event.key], query);
+		case 'update':
+			return await service.readMany(event.keys, query);
+		case 'delete':
+			return event.keys;
```

Comment:
Should we add a default case here to cover for potential addition of event actions?